### PR TITLE
createNamespace when connecting to Stargate

### LINKED ISSUE #11: Create keyspace while connecting
For a locally hosted JSON API Server (instead of AstraDB), it will be helpful to create a keyspace while connecting. https://github.com/stargate/sg-mongoose2/pull/7#discussion_r1090925049

### PR DESCRIPTION
**Pull request by vkarpov15**
**What this PR does**:

Run the `createNamespace` command to create a new namespace (keyspace) when connecting, just in case the namespace doesn't already exist. Because without the namespace, all operations will fail.

It doesn't look like `createNamespace` throws an error if the namespace already exists, so don't need to check that.

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [X] Changes manually tested
- [ ] Automated Tests added/updated
- [ ] Documentation added/updated
- [ ] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Review by kathirsvn — APPROVED** [review]
LGTM

**Comment by kathirsvn** [comment]
Hi @vkarpov15 ,
I have improved this solution after my testing on local. This takes care of below. Could you please review and approve? 

- The test client created for `npm test` also creates Namespace as required (this doesn't use the `OpenUri` function in `driver/connection.ts`, instead this client calls the static `connect` function of `Client` directly)
- URI parsing is fixed (for keyspace name), so we don't have to fix the keyspace name (i.e. replacing /v1 with empty string). Basically, in the following URL for example:`http://localhost:8080/v1/testks1?applicationToken=914619c4-df79-4294-87e7-d7fe9a2ff00d&baseApiPath=v1`, below are the components
```
baseUrl: http://localhost:8080
baseApiPath: v1
keypspaceName: testks1
applicationToken=914619c4-df79-4294-87e7-d7fe9a2ff00d
```
- Added a client option called `createNamespaceOnConnect` to check whether to create Namespace on connection or not. (this also takes care of some of the test cases where we pass invalid connection params in our tests, in which we don't want to create Namespace automatically)

File: src/collections/utils.ts [hunk 1]
```diff
@@ -36,7 +38,7 @@ interface ParsedUri {
 export const parseUri = (uri: string): ParsedUri => {
   const parsedUrl = url.parse(uri, true);
   const baseUrl = `${parsedUrl.protocol}//${parsedUrl.host}`;
-  const keyspaceName = parsedUrl.pathname?.replace('/', '');
+  const keyspaceName = parsedUrl.pathname?.substring(parsedUrl.pathname?.lastIndexOf('/') + 1)
```
Comment (vkarpov15) [inline src/collections/utils.ts:41]: Confirmed that this works fine even if there's no '/' in `parsedUrl.pathname`, apparently `str.substring(NaN)` is equivalent to `str.substring(0)`.

**Review by vkarpov15 — COMMENTED** [review]
Works great, thanks @kathirsvn , LGTM on my end. Please review and merge this when you're ready.

**Comment by kathirsvn** [comment]
Thanks @vkarpov15 Let me resolve the conflicts and update again.

**Pull request merged**